scripts/dts/gen_defines.py

```diff
@@ -476,29 +476,41 @@ def write_phandle_val_list(dev, entries, ident):
     #
     # entries:
     #   List of entries (two for 'pwms' above). This might be a list of
-    #   edtlib.PWM instances, for example.
+    #   edtlib.PWM instances, for example.  If only one entry is given it
+    #   does not have a suffix '_0', and the '_COUNT' and group initializer
+    #   are not emitted.
     #
     # ident:
     #   Base identifier. For example, "PWM" generates output like this:
     #
     #     #define <device prefix>_PWMS_CONTROLLER_0 "PWM_0"  (name taken from 'label = ...')
     #     #define <device prefix>_PWMS_CHANNEL_0 123         (name taken from #cells in binding)
+    #     #define <device prefix>_PWMS_0 {"PWM_0", 123}
     #     #define <device prefix>_PWMS_CONTROLLER_1 "PWM_1"
     #     #define <device prefix>_PWMS_CHANNEL_1 456
+    #     #define <device prefix>_PWMS_1 {"PWM_1", 456}
+    #     #define <device prefix>_PWMS_COUNT 2
+    #     #define <device prefix>_PWMS {<device_prefix>_PWMS_0, <device_prefix_PWMS_1}
     #     ...
     #
     #   Note: Do not add an "S" to 'ident'. It's added automatically, which
     #   forces consistency.
 
+    init_vals = []
     for i, entry in enumerate(entries):
-        write_phandle_val_list_entry(
-            dev, entry, i if len(entries) > 1 else None, ident)
+        init_vals.append(write_phandle_val_list_entry(
+            dev, entry, i if len(entries) > 1 else None, ident))
+    if len(entries) > 1:
```

Comment:
Might be nice to streamline in and always do the same thing, unless it's helpful to special-case it for a single entry.

Reply:
Can't.  You get redefined macros because there's no `_0` suffix on the initializer for the single item, to distinguish it from the initializer for multiple items:
```
 #define DT_GPIO_LEDS_LED_0_GPIOS                    {"GPIO_1", 12, 0}
+#define DT_GPIO_LEDS_LED_0_GPIOS_COUNT              1
+#define DT_GPIO_LEDS_LED_0_GPIOS                    {DT_GPIO_LEDS_LED_0_GPIOS}
```

Reply:
So it's that old inconsistency forcing more inconsistency? Fine then, unless you come up with a different name or something.

This is why I don't like stuff like that, and think it's super important to fix it quickly. Messes always grow other messes.

Reply:
Agreed.  But there are `foo-gpios` that denote something that will never be multiple, and `foo-gpios` that denote things that will always be (potentially) multiple.  Unless the type distinguishes them I don't see an alternative.

I think I've gotten all requested changes now (missed a couple in first push).